### PR TITLE
Registration Form

### PR DESCRIPTION
**Pull request by talyh**
Basic form structure and behaviour
Data saving and loading

**Coming Up Next**
Show Confirmation message on form submit
Sign Out
Redux

### CODE REVIEW

File: public/manifest.json [hunk 1]
```diff
@@ -1,23 +1,7 @@
 {
```
Comment (rota-rossi) [inline public/manifest.json]: this file is also only used for PWA (WebWorker). you can safely remove it.

File: src/components/formElements/FormArea.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      return "#c02549";
+    } else {
+      return "#a4a4a4";
+    }
+  }};
+  background-color: ${props => {
+    if (props.active) {
+      return "#2569c0";
+    } else if (props.error) {
+      return "#c02549";
```
Comment (rota-rossi) [inline src/components/formElements/FormArea.tsx]: consider moving the colour codes to an external file, and import them from there. makes it easier to maintain consistency.

File: src/components/formElements/OptionsGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  valuesArray,
+  required = false,
+  height,
+  shouldDisable
+}: IOptionGroupProps) => {
+  const { meta } = useField(name);
+
+  return (
+    <FormArea height={height} error={meta.error}>
+      <Label required={required} style={{ top: 0 }}>
```
Comment (rota-rossi) [inline src/components/formElements/OptionsGroup.tsx]: Is this inline style necessary?

File: src/views/RegistrationForm/RegistrationForm.tsx [hunk 4]
```diff
@@ -1,13 +1,95 @@
...
+      <FormContainer>
+        <Form
+          initialValues={user as IUser}
+          onSubmit={async (data: IUser) => {
+            await saveUserData(uid, data);
+            setUser(data);
+          }}
+          validate={validate}
+          render={({ errors, handleSubmit, submitting }) => (
+            <form onSubmit={handleSubmit}>
```
Comment (rota-rossi) [inline src/views/RegistrationForm/RegistrationForm.tsx]: Is this form inside the `Form` Component necessary?

**Comment by rota-rossi** [comment]
How about better commit messages? 😃

File: src/components/formElements/FormArea.tsx (hunk 2 above)
Comment (talyh) [reply src/components/formElements/FormArea.tsx]: Will look into it for a future PR

File: src/components/formElements/OptionsGroup.tsx (hunk 3 above)
Comment (talyh) [reply src/components/formElements/OptionsGroup.tsx]: Not absolutely, but ti does look better with it, than without, and it's only really needed in this instance. I could make it a prop for FormArea, but seems overkill.

File: src/views/RegistrationForm/RegistrationForm.tsx (hunk 4 above)
Comment (talyh) [reply src/views/RegistrationForm/RegistrationForm.tsx]: Yes. It's how the library is implemented.
https://github.com/final-form/react-final-form/blob/master/examples/simple/index.js

**Comment by talyh** [comment]
> How about better PR Names? 😃

[Changes PR name]
I have no idea what you're talking about! =D

File: public/manifest.json (hunk 1 above)
Comment (talyh) [reply public/manifest.json]: 👍 
In a future update

**Pull request merged**